### ISSUE TITLE
MQTT error [-1] when using ipv6 URI

### ISSUE THREAD

**Issue by bmulder-innoseis** [eclipse/paho.mqtt.cpp#212]
When I initialize a connection using an ipv6 URI (in the form of e.g. tcp://[1234::5678:9abc]:1883) I get "MQTT error [-1]:" when I call connect on the client. 

Maybe this is related to [https://github.com/eclipse/paho.mqtt.c/issues/179](https://github.com/eclipse/paho.mqtt.c/issues/179) ?

Anyway, is ipv6 supposed to be supported?

**Comment by fpagliughi** [comment]
Yes, the library should support IPv6 sockets/addresses. The C++ library just passes the server URI string to the underlying C lib. I'll test it out to verify that it works.

**Comment by fpagliughi** [comment]
We're hoping for IPv6 issues to be resolved in the upcoming Paho C v1.4 release...

**Comment by fpagliughi** [comment]
The upcoming Paho C release, v1.3.14, add improvements to IPv6 support, but I can't tell if it's 100% yet.
